### ISSUE TITLE
Internalization: Add your local language

### ISSUE THREAD

**Issue by aboullaite** [fbdevelopercircles/open-source-edu-bot#57]
#### Is your feature request related to a problem? Please describe.
The bot interact with users with english. We can customize the user experience by adding support for i18n. If you're interested in adding support for a language, please follow the steps below:

1. **Determine the language code**  
The `ISO 639-1` standard defines two-letter codes for languages. Find the two-letter codes of the language you want to add [here](http://en.wikipedia.org/wiki/List_of_ISO_639-1_codeshttp://en.wikipedia.org/wiki/List_of_ISO_639-1_codes). This document assumes that you chose ‘fr’ for French.
2. **Create .po file (One-time only)**
.po translation files are generated from .pot template files using pybabel init.
```bash
            $ cd ~/path/to/open-source-edu-bot
(venv) $ pybabel init -d locales -l fr -i messages.pot
```
3. **Add your changes**
got to the newly generated file and start i18n process. you can get inspired from `src/locales/fr/LC_MESSAGES/messages.po` 
4. **Translate .po file**
Compile a .mo compiled translation file from a .po translation file like so:
```bash
pybabel compile -d locales
```
5. **Verify**
Run the app in your test en, and verify that the bot answers you in your selected language

**Comment by ChiragTutlani** [comment]
I will take up of translating it to hindi (ISO 639-1 code: hi).

**Comment by aboullaite** [comment]
Awesome @ChiragTutlani 🎉 please do not hesitate to reach oit if you need any assistance 👌

**Comment by elinguiuriel** [comment]
@ChiragTutlani I made an update to describe how to translate ``GREETINGS` and `PERSISTANT_MENU` take a look [here](https://github.com/fbdevelopercircles/open-source-edu-bot/issues/57#issue-666934319)

**Comment by prilcool** [comment]
I added Simplified Chinese  (zh-Hans)   work in progress   in the locales folder , 
Welcome to  do a pull and  continue translation  after its merged  :)

**Comment by ankitaggarwal23** [comment]
i want to contribute in this issue

**Comment by SrilalS** [comment]
Hi! I Would like to Take Translation to Sinhala (Code: si).

**Comment by vj-codes** [comment]
I would like to add German (code:de) translation!

**Comment by elinguiuriel** [comment]
> I added Simplified Chinese (zh-Hans) work in progress in the locales folder ,
> Welcome to do a pull and continue translation while i'm asleep lol :)

Are you a chinesse speaker ? Please only add a language if you are a native speaker.

**Comment by elinguiuriel** [comment]
> I would like to add German (code:de) translation!

Are you a german speaker ?

**Comment by elinguiuriel** [comment]
> Hi! I Would like to Take Translation to Sinhala (Code: si).

Great, first make sure the langage is supporter https://developers.facebook.com/docs/messenger-platform/messenger-profile/supported-locales
are you a native speaker or that language, or Can you have help from a native speaker ?

**Comment by aboullaite** [comment]
@elinguiuriel being a native speaker is not necessary IMHO! Anyone who can correctly translate the messages is welcomed to open a PR as far is he's capable of completing it! Anyone who find issues on translation can submit improvements afterwards

**Comment by vj-codes** [comment]
> > I would like to add German (code:de) translation!
> 
> Are you a german speaker or do you know a good german speaker willing to help you ?

I'm a speaker , I know german as an additional language.If required I can ask a native speaker for PR review too. @elinguiuriel

**Comment by SrilalS** [comment]
@aboullaite I Began the translation process for the Sinhala (Si). I have one question. there are some English words that have native words but those are less used and not understood by General Public. Should I made the translations in Pure Sinhala or Can I do it in Modern Sinhala?

**Comment by SrilalS** [comment]
> > Hi! I Would like to Take Translation to Sinhala (Code: si).
> 
> Great, first make sure the langage is supporter https://developers.facebook.com/docs/messenger-platform/messenger-profile/supported-locales
> are you a native speaker or that language, or Can you have help from a native speaker ?

I'm a Native Speaker.

**Comment by aboullaite** [comment]
@vj-codes please continue your amazing work 🙏
@SrilalS the goal is to be understandable, easy to use and help users. I think modern Sinhala is better base on what you described

**Comment by elinguiuriel** [comment]
> @elinguiuriel being a native speaker is not necessary IMHO! Anyone who can correctly translate the messages is welcomed to open a PR as far is he's capable of completing it! Anyone who find issues on translation can submit improvements afterwards

Just we have to make sure to not sacrifice on quality when pushing translation in production. Because we have no way of controling translation quality ! If someone is not fluent enough in a language I prefer that his translation should be validate by a fluent speakers before being merged.

**Comment by elinguiuriel** [comment]
> > > I would like to add German (code:de) translation!
> > 
> > Are you a german speaker or do you know a good german speaker willing to help you ?
> 
> I'm a speaker , I know german as an additional language.If required I can ask a native speaker for PR review too. @elinguiuriel 

I am a german speaker too, but a very bad speaker. If you can find someone fluent to validate the first version, it will be great.
I ask this question because it is a lot of work, there is a lot of content in the app and being fluent help a lot.

**Comment by aboullaite** [comment]
@elinguiurie, faor enough,l we could require review from devc community before merging! But bring a native speaker should not be the only criteria to suggest a PR 👍

**Comment by aayushbisen** [comment]
#87

**Comment by elinguiuriel** [comment]
> @elinguiurie, faor enough,l we could require review from devc community before merging! But bring a native speaker should not be the only criteria to suggest a PR 👍

Fluent or Native Speaker

**Comment by Yannix** [comment]
> 
> 
> @ChiragTutlani I made an update to describe how to translate `GREETINGS` and `PERSISTANT_MENU` take a look [here](https://github.com/fbdevelopercircles/open-source-edu-bot/issues/57#issue-666934319)

French translation working for the menu but not working inside the bot. Seems that the translation process not applied while in the chat.

**Comment by elinguiuriel** [comment]
> > 
> > 
> > @ChiragTutlani I made an update to describe how to translate `GREETINGS` and `PERSISTANT_MENU` take a look [here](https://github.com/fbdevelopercircles/open-source-edu-bot/issues/57#issue-666934319)
> 
> French translation working for the menu but not working inside the bot. Seems that the translation process not applied while in the chat.

I suspect that your Facebook language is not "Français (France)" or any other version or french, and your phone is localised in french. Make sure you have french everywhere

![Screenshot_20200802-182652_Messenger.jpg](https://user-images.githubusercontent.com/2127223/89129560-c2471f00-d4ed-11ea-8295-ccb37255035d.jpg)

**Comment by prilcool** [comment]
> > @elinguiuriel being a native speaker is not necessary IMHO! Anyone who can correctly translate the messages is welcomed to open a PR as far is he's capable of completing it! Anyone who find issues on translation can submit improvements afterwards
> 
> Just we have to make sure to not sacrifice on quality when pushing translation in production. Because we have no way of controling translation quality ! If someone is not fluent enough in a language I prefer that his translation should be validate by a fluent speakers before being merged.

This is not all or nothing situation .
You can control the quality of what's being merged into the production branch .
You should accept submissions on the development branch
Than next invite the Global Devc community to vet the translations and have a vote on it , after all thats what the Devc Community is about .
I believe that would be a formal ,fun and democratic  way of  doing things.

**Comment by elinguiuriel** [comment]
> > > @elinguiuriel being a native speaker is not necessary IMHO! Anyone who can correctly translate the messages is welcomed to open a PR as far is he's capable of completing it! Anyone who find issues on translation can submit improvements afterwards
> > 
> > 
> > Just we have to make sure to not sacrifice on quality when pushing translation in production. Because we have no way of controling translation quality ! If someone is not fluent enough in a language I prefer that his translation should be validate by a fluent speakers before being merged.
> 
> This is not all or nothing situation .
> You can control the quality of what's being merged into the production branch .
> You should accept submissions on the development branch
> Than next invite the Global Devc community to vet and correct the translations and have a vote on it , after all thats what the Devc Community is about .
> I believe that would be a formal ,fun and democratic way of doing things.

@prilcool you have a great point here! Please feel free to translate any language you are fluent in. The translation is a small part of the project, and it is not the easiest, it's quite a time-consuming and error-prone. Because we are a large global community as you noted, I thought it will be better to let such specialized tasks to people most able to understand their translation, and not waste precious contributors time doing work that will not be merged or that will require too many corrections.

**Comment by aboullaite** [comment]
@prilcool if you read carefully the discussion here, you should mention that exactly what I was advocating for and what we agreed to do from now on. Relying solely on GT for translating to other languages will generate a negative experience as mentioned by @elinguiuriel

**Comment by prilcool** [comment]
@aboullaite  @elinguiuriel   Ah  i see now, after testing with other language  apart from Chinese, for some language   gt translate  is not so good and funny  ok we leave translation to native speakers .

**Comment by Emile-Nsengimana** [comment]
I would like to add Kinyarwanda (code:rw) translation!

**Comment by pixan198** [comment]
I would like to translate it to Gujarati (India) `ISO 639-1 code: gu`

**Comment by LuisFerTR** [comment]
Hi, I would like to add spanish (code: es) translation!

**Comment by elinguiuriel** [comment]
> Hi, I would like to add spanish (code: es) translation!

@LuisFerTR please yourself 👍

**Comment by shah78677** [comment]
I would like to add Urdu language. ur_PK

**Comment by stonefix** [comment]
Hi! I am a native Russian speaker and I would like to add Russian language.

**Comment by elinguiuriel** [comment]
> Hi! I am a native Russian speaker and I would like to add Russian language.

Please yourself @stonefix the russian code is `ru_RU` according to [this](https://developers.facebook.com/docs/messenger-platform/messenger-profile/supported-locales)

**Comment by shubhankarsharma876** [comment]
> > Hi! I am a native Russian speaker and I would like to add Russian language.
> 
> Please yourself @stonefix the russian code is `ru_RU` according to [this](https://developers.facebook.com/docs/messenger-platform/messenger-profile/supported-locales)

Hey thanks @elinguiuriel for attaching that list.

**Comment by shubhankarsharma876** [comment]
I will add Kannada (INDIA) transcript with code ```kn_IN``` according to [this](https://developers.facebook.com/docs/messenger-platform/messenger-profile/supported-locales)

**Comment by shubhankarsharma876** [comment]
This is my first contribution to any project and I know this could be a silly question, But...
I have a query
 (venv) $ curl -X GET "<YOUR HOST>/webhook?hub.verify_token=<YOUR VERIFY TOKEN>&hub.challenge=CHALLENGE_ACCEPTED&hub.mode=subscribe&init_bot=true"

what do I have to put in place of "YOUR HOST" and "YOUR VERIFY TOKEN"?
And do I have to run this code before starting contribution?

**Comment by shubhankarsharma876** [comment]
I ran this code [ (venv) $ pybabel extract -F babel.cfg -k lazy_gettext -o locales/messages.pot .] in my git but showed the below error.

Please provide me some fixes: I installed the anaconda and python 3.10.2 properly with the path. But still getting this error

$ pybabel extract -F babel.cfg -k lazy_gettext -o locales/messages.pot .        Traceback (most recent call last):
  File "C:\Users\shubh\anaconda3\Scripts\pybabel-script.py", line 9, in <module>
    sys.exit(main())
  File "C:\Users\shubh\anaconda3\lib\site-packages\babel\messages\frontend.py", line 929, in main
    return CommandLineInterface().run(sys.argv)
  File "C:\Users\shubh\anaconda3\lib\site-packages\babel\messages\frontend.py", line 853, in run
    return cmdinst.run()
  File "C:\Users\shubh\anaconda3\lib\site-packages\babel\messages\frontend.py", line 438, in run
    mappings = self._get_mappings()
  File "C:\Users\shubh\anaconda3\lib\site-packages\babel\messages\frontend.py", line 502, in _get_mappings
    with open(self.mapping_file, po_file_read_mode) as fileobj:
FileNotFoundError: [Errno 2] No such file or directory: 'babel.cfg'
